signingscript/test/integration/test_autograph.py

```diff
@@ -217,25 +207,50 @@ def _extract_compress_type_per_filename(path):
     makedirs(copied_file_folder)
     shutil.copy(original_file_path, copied_file_folder)
 
-    zip_infos_before_signature = _extract_compress_type_per_filename(os.path.join(copied_file_folder, file_name))
-
     context.config['signing_server_config'] = _write_server_config(tmpdir)
     context.task = _craft_task([file_name], signing_format='autograph_focus')
 
     keystore_path = os.path.join(tmpdir, 'keystore')
     certificate_path = os.path.join(TEST_DATA_DIR, 'autograph_apk.pub')
     certificate_alias = 'autograph_focus'
-    _instanciate_keystore(keystore_path, certificate_path, certificate_alias)
+    _instantiate_keystore(keystore_path, certificate_path, certificate_alias)
 
     await async_main(context)
 
-    signed_path = os.path.join(tmpdir, 'artifact', file_name)
-    assert _verify_apk_signature(keystore_path, signed_path, certificate_alias)
+def _extract_apk_signature_algorithm(apk_path):
+    cmd = [
+        _get_java_path('jarsigner'), '-verify', '-verbose',
+        apk_path
+    ]
+    log.info("running {}".format(cmd))
+    command = subprocess.run(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines=True
+    )
+
+    assert command.returncode == 0
+
+    algorithm_line = [line for line in command.stdout.split('\n') if 'Signature algorithm:' in line][0]
+    algorithm = algorithm_line.strip().split(' ')[2]
+    return algorithm[:-1]  # remove comma at end of algorithm
+
+
+@pytest.mark.asyncio
+@skip_when_no_autograph_server
+@pytest.mark.parametrize('format,expected_algorithm', ((
+    'autograph_apk_fennec_sha1', 'SHA1withSHA1withRSA',
+), (
+    'autograph_apk_fennec', 'SHA256withSHA256withRSA',
+)))
+async def test_integration_autograph_custom_digest_algorithm(context, tmpdir, format, expected_algorithm):
+    file_name = 'app.apk'
+    original_file_path = os.path.join(TEST_DATA_DIR, file_name)
+    shutil.copy(original_file_path, tmpdir)
+    apk_path = os.path.join(tmpdir, file_name)
 
-    zip_infos_after_signature = _extract_compress_type_per_filename(signed_path)
-    for signature_file in ('META-INF/SIGNATURE.RSA', 'META-INF/SIGNATURE.SF', 'META-INF/MANIFEST.MF'):
-        del zip_infos_after_signature[signature_file]
+    context.signing_servers = {'project:releng:signing:cert:dep-signing': [SigningServer(
+        *["http://localhost:5500", "bob", "1234567890abcdefghijklmnopqrstuvwxyz1234567890abcd", [format], "autograph"]
+    )]}
+    context.task = _craft_task([file_name], signing_format=format)
 
-    # We want to make sure compression type hasn't changed after the signature
-    # https://github.com/mozilla-services/autograph/issues/164
-    assert zip_infos_before_signature == zip_infos_after_signature
```

Comment:
We probably want to keep this test. It caused https://bugzilla.mozilla.org/show_bug.cgi?id=1513870#c22 last week.

Reply:
:grimacing: that's a huge :yikes: from me, I'm not sure how this got removed.
Great eye, and I promise it wasn't deliberate (is that worse?)

Reply:
Don't blame yourself. This is why code reviews exist :)